server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -678,6 +685,23 @@ public synchronized IndexShard createShard(
         }
     }
 
+    // When an instance of Store is created, a shardlock is created which is released on closing the instance of store.
+    // Currently, we create 2 instances of store for remote store backed indices: store and remoteStore.
+    // As there can be only one shardlock acquired for a given shard, the lock is shared between store and remoteStore.
+    // This creates an issue when we are deleting the index as it results in closing both store and remoteStore.
+    // Sample test failure: https://github.com/opensearch-project/OpenSearch/issues/13871
+    // The following method provides ShardLock that is not maintained by NodeEnvironment.
+    // As part of https://github.com/opensearch-project/OpenSearch/issues/13075, we want to move away from keeping 2
+    // store instances.
+    private ShardLock getRemoteStoreLock(ShardId shardId) {
+        return new ShardLock(shardId) {
+            @Override
+            protected void closeInternal() {
+                // Ignore for remote store
+            }
+        };
+    }
```

Comment:
Each invocation of `getRemoteStoreLock()` will create a new `ShardLock` for the given `shardId`. 

Should this method instead return the existing `ShardLock` for the `shardId` (if it is still valid), else return a new one? 

I think the logic here should be robust enough to avoid hard-to-trace errors due to inadvertently invoking this method multiple times for the same shardId. The current logic here is _creational_ rather than just _getting_ (i.e. what the name suggests) or _getOrCreate_.

Reply:
Agree, let me make the required changes.

Reply:
Moved shard lock creation inline, please check.